pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -618,12 +618,14 @@ public String getCrypterClassNameFromTableConfig(String tableNameWithType) {
       }
     }
     // Fetch the segment lineage metadata, and filter segments based on segment lineage.
-    ZNRecord segmentLineageZNRecord =
-        SegmentLineageAccessHelper.getSegmentLineageZNRecord(_propertyStore, tableNameWithType);
-    SegmentLineage segmentLineage = SegmentLineage.fromZNRecord(segmentLineageZNRecord);
-    Set<String> selectedSegmentSet = new HashSet<>(selectedSegments);
-    SegmentLineageUtils.filterSegmentsBasedOnLineageInPlace(selectedSegmentSet, segmentLineage);
-    return new ArrayList<>(selectedSegmentSet);
+    SegmentLineage segmentLineage = SegmentLineageAccessHelper.getSegmentLineage(_propertyStore, tableNameWithType);
+    if (segmentLineage == null) {
```

Comment:
We don't need to treat `null` separately?

`SegmentLineageUtils.filterSegmentsBasedOnLineageInPlace()` has the null check so `null` case is covered. 

The output with/without this `if (segmentLineage==null)` is the same. So we can remove this if statement.

Reply:
The benefit of having the if check here is to reduce the collection instantiation, ie. we don't have to instantiate one more hash set and then another list if the segment lineage doesn't exist. We should be good here.